app/src/main/java/edu/wisc/ece/pinpoint/pages/profile/EditProfileFragment.java

```diff
@@ -202,4 +230,57 @@ private void launchLocationAutocomplete(View view, boolean isFocused) {
                             requestedFields).setTypesFilter(placesFilter).build(requireContext()));
         }
     }
+
+    private void takePicture() {
+
+        Intent pictureIntent = new Intent(
+                MediaStore.ACTION_IMAGE_CAPTURE).addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        if(pictureIntent.resolveActivity(this.getContext().getPackageManager()) != null){
+            //Create a file to store the image
+            try {
+                photoFile = createImageFile();
+            } catch (IOException ex) {
+                Toast toast = Toast.makeText(this.getContext(), getString(R.string.file_creation_failed_message), Toast.LENGTH_SHORT);
```

Comment:
Replace `this.getContext()` with `requireContext()` to prevent NullPointerExceptions